filter.go

```diff
@@ -66,3 +76,112 @@ func (cf CorrelationFilter) ToFilterDescription() FilterDescription {
 		CorrelationFilter: cf,
 	}
 }
+
+// MarshalXML implements the xml.Marshaller interface for CorrelationFilterProperties.
+func (c CorrelationFilterProperties) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
+	if len(c) == 0 {
+		return nil
+	}
+	if err := e.EncodeToken(start); err != nil {
+		return err
+	}
+	keyvalWrapper := xml.StartElement{Name: xml.Name{Local: "KeyValueOfstringanyType"}}
+	for k, v := range c {
+		if err := e.EncodeToken(keyvalWrapper); err != nil {
+			return err
+		}
+		if err := e.EncodeElement(k, xml.StartElement{Name: xml.Name{Local: "Key"}}); err != nil {
+			return err
+		}
+		if err := e.EncodeElement(v, xml.StartElement{Name: xml.Name{Local: "Value"}}); err != nil {
+			return err
+		}
+		if err := e.EncodeToken(keyvalWrapper.End()); err != nil {
+			return err
+		}
+	}
+	return e.EncodeToken(start.End())
+}
+
+// UnmarshalXML implements the xml.Unmarshaller interface for CorrelationFilterProperties.
+func (c *CorrelationFilterProperties) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	var key string
+	var val interface{}
+	*c = map[string]interface{}{}
```

Comment:
Just double checking, there's no case where c could be nil right?

Reply:
Correct.  The unmarshaller will create an instance of that for us.